handlers/src/main/java/com/nextdoor/bender/handler/kinesis/KinesisEventIterator.java

```diff
@@ -54,7 +69,19 @@ public boolean hasNext() {
 
   @Override
   public InternalEvent next() {
-    return new KinesisInternalEvent(this.iterator.next(), this.context, this.shardId);
+    KinesisEventRecord nextRecord = this.iterator.next();
+    if (assumeKinesisDataIsGzipped) {
+      ByteBuffer gzip = nextRecord.getKinesis().getData();
+      try (GZIPInputStream gzipInputStream = new GZIPInputStream(new ByteArrayInputStream(gzip.array()))) {
+        IOUtils.copy(gzipInputStream, byteArrayOutputStream, bufferSize);
+        nextRecord.getKinesis().setData(ByteBuffer.wrap(byteArrayOutputStream.toByteArray()));
+      } catch (Exception e) {
```

Comment:
Be explicit with what errors we're expecting here. The blanket `Exception` should only be used sparingly.

Reply:
changed to IOException